rust/src/fedora_integration.rs

```diff
@@ -204,19 +218,69 @@ fn is_debug_rpm(rpm: &str) -> bool {
     rpm.contains("-debuginfo-") || rpm.contains("-debugsource-")
 }
 
-pub(crate) fn handle_cli_arg(url: &str, arch: &str) -> CxxResult<Option<Vec<File>>> {
+fn get_base_package_list() -> Result<HashSet<String>> {
+    if crate::running_in_container() {
+        Ok(
+            crate::ffi::rpmdb_package_name_list(libc::AT_FDCWD, "/".to_string())?
+                .into_iter()
+                .collect(),
+        )
+    } else {
+        let sysroot = &ostree::Sysroot::new_default();
+        sysroot.load(gio::NONE_CANCELLABLE)?;
+        let deployments = sysroot.deployments();
+        let default = deployments
+            .get(0)
+            .ok_or_else(|| anyhow::anyhow!("No deployments found"))?;
+        let checksum = default.csum().unwrap();
+        let repo = sysroot.repo().unwrap();
+        let repo = repo.gobj_rewrap();
+        let pkglist = {
+            let cancellable = gio::Cancellable::new();
+            unsafe {
+                let r = crate::ffi::package_variant_list_for_commit(
```

Comment:
This part isn't unsafe right?

Reply:
https://github.com/coreos/rpm-ostree/pull/3705